pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -859,6 +852,13 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 				return volumeToMount.GenerateError("MapVolume failed", fmt.Errorf("Device path of the volume is empty"))
 			}
 		}
+		// Update actual state of world to reflect volume is globally mounted
```

Comment:
somehow my comment is not showing up.
My only concern is by moving actualStateOfWorld.MarkDeviceAsMounted() later, if any failure happened above it, actualstate will not have a record of this globalMapPath. 
Maybe it will be more clear if the code can show where part is for attachable and which part is for non-attacable, and when we should update the actual state.